r/R/record-batch.R

```diff
@@ -189,3 +189,62 @@ record_batch <- RecordBatch$create
 
 #' @export
 names.RecordBatch <- function(x) x$names()
+
+#' @export
+rbind.RecordBatch <- function(...) {
+  abort("Use `Table$create()` to combine record batches")
+}
+
+cbind_check_length <- function(target_length, length, idx, call = caller_env()) {
```

Comment:
Do you need this function? I'm pretty sure the RecordBatch constructor in C++ will check lengths (again).

Reply:
The main value of the function is to provide an error message indicating which of you inputs is not the correct length. But if we didn't care about that, we could rely on the RB constructor.

Reply:
Any reason the RB constructor couldn't tell you which input was the wrong length?

Reply:
RB constructor inputs are always arrays, right? Whereas here, the first input is guaranteed to be a record batch. So there's not a 1:1 correspondence between arguments there that would let us provide a helpful error message.